### ISSUE TITLE
Error: File Not Processing

### ISSUE THREAD

**Issue by DJMike67** [AzuraCast/AzuraCast#3263]
Hi there. I run a radio station but don't run my own server so unfortunately I can't provide too many details such as use of Docker, operating system etc. Although at bottom of the screen it says Rolling Release #7de1012 (2020-10-07 9:10) • Docker • PHP 7.4

For several days I have been uploading MP3 and M4A files via FTP (web interface wouldn't work). The files upload successfully and appear in the correct folder in the web interface but some of them have "File Not Processed" just above the filename in the folder list

There is no consistency
3 small files (less than 2MB each) processed with no problem
1 file of 63MB processed successfully
However 2 files of 60MB did not process

It is now several days and they still haven't processed 

I also get the same problem (File Not Processed message) if I try and move a file to a different folder via the web interface 

If anyone has any suggestions I would be extremely grateful. It renders the AutoDJ feature useless if i can only play existing files hence hampering the output of my radio station

Many thanks

**Comment by SlvrEagle23** [comment]
@DJMike67 What error are you seeing when you move or try to upload files via the web interface?

You may need to check your browser's console (usually the F12 key), click the "Network" tab, then try the action again and look for a request that lights up in red, click it and then copy/paste its response body (the details of the error) back to us here.

Your release version is relatively new, but also lands in the middle of some large development work we've been doing, so it's possible there was an error in that specific version that updating would fix.

**Comment by DJMike67** [comment]
@SlvrEagle23 In my testing, uploading a small (2MB) file via web interface or FTP it works (I don't get the File Not Processed message). However with a larger file (60MB), it appears to upload successfully via the web interface - i.e. the progress bar gets to the end, but the file does not appear in the folder. I have left it several hours

If I move a file to another folder, via the web interface, I get a message that says the file has been moved. However the file does not actually move. It remains in the original folder with the message "File Not Processed". this is true for all files, regardless of size.

When I go into the browser console and copy /paste the response, this is what I get

<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">

    <title>Error</title>

    <link rel="apple-touch-icon" sizes="180x180" href="/static/icons/production/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/static/icons/production/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/static/icons/production/favicon-16x16.png">
<link rel="manifest" href="/static/icons/production/site.webmanifest">
<link rel="mask-icon" href="/static/icons/production/safari-pinned-tab.svg" color="#5bbad5">
<link rel="shortcut icon" href="/static/icons/production/favicon.ico">
<meta name="msapplication-TileColor" content="#2196F3">
<meta name="msapplication-config" content="/static/icons/production/browserconfig.xml">
<meta name="theme-color" content="#2196F3">

    <link rel="stylesheet" type="text/css" href="/static/dist/lib/roboto-fontface/css/roboto/roboto-fontface-691f1edc86.css" />
<link rel="stylesheet" type="text/css" href="/static/dist/lib/material-icons/material-icons-6b78e76e7f.css" />
<link rel="stylesheet" type="text/css" href="/static/dist/light-3e10aea1a5.css" />
<style type="text/css" nonce="uBk2qSEWVsYZzdLziVTp6D0X">
body.page-minimal {
    background: #edecec url('#') center center;
    background-repeat: no-repeat;
    background-color: black;
    background-size: contain;
}</style>
<script type="text/javascript" src="/static/dist/lib/jquery/jquery-dc5e7f18c8.min.js"></script>
<script type="text/javascript" src="/static/dist/lib/bootstrap/bootstrap-21f815ff6d.bundle.min.js"></script>
<script type="text/javascript" src="/static/dist/lib/bootstrap-notify/bootstrap-notify-a02f92a499.min.js" defer></script>
<script type="text/javascript" src="/static/dist/app-26ad9f229a.js" defer></script>
<script type="text/javascript" src="/static/dist/material-df68dbf23f.js"></script>
</head>

<body class="page-minimal error-content theme-light">
<script type="text/javascript" nonce="uBk2qSEWVsYZzdLziVTp6D0X">
let App = {"lang":{"confirm":"Are you sure?","placeholder":"Select...","no_results":"No results found!","advanced":"Advanced"},"locale":"en_US","locale_with_dashes":"en-US"};</script>


<div class="public-page">
    <div class="card">
        <div class="card-body">
            <h2 class="display-1 mt-2">500</h2>
            <p>
                An exception occurred while executing 'UPDATE station_media SET path = ? WHERE id = ?' with params ["Brooklands196 Specials\/", 45984]:

SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'Brooklands196 Specials/-9' for key 'path_unique_idx'<br>
                AbstractMySQLDriver.php : 74            </p>

            <a role="button" title="Dashboard" class="btn btn-login btn-primary btn-float" href="/">
                <i class="material-icons" aria-hidden="true">home</i>
            </a>
        </div>
    </div>
</div>


</body>
</html>

**Comment by DJMike67** [comment]
@SlvrEagle23 I have an update to  this issue (#3263). But also this relates to #3270 which another user submitted

It appears that the company that I use for my streaming service updated to Rolling Release #b31f892 (2020-10-13 4:50)

Prior to this update, when I uploaded files through FTP, the small file (under 2MB) processed successfully within a few minutes. The larger files (60MB) did not process and a message appeared "File Not Processed")

I have today tested again by uploading several files (small and large) via FTP but now ALL those files display "File Not Processed" in the name column. I have waited over 1 hour and they still haven't processed.

However, if I select the files and move them to another folder, 3 of them move without an error and magically the file not processed message disappears from these 3 files

However, with several other files, when I attempt to move them to another folder I get a message "An error occurred and your request could not be completed" and it continues to show "File Not Processed". 

I have tried several times with these files - deleting it and re-uploading it. There is nothing wrong with the MP3's themselves - they play locally via Quicktime player and also play on another radio station that uses Centovcast. Seems to be just AzuraCast it doesn't like!

Please can you provide me with an ETA for a fix. This is rendering the AutoDJ element of my radio station useless and I am having to consider moving away from Azuracast

**Comment by SlvrEagle23** [comment]
@DJMike67 In order to even begin working on a fix we would need additional information about the specific errors you're running into in this case. A majority of our users aren't having mass problems processing media files, so we need to isolate what's different about your installation that's causing the problem; the more logs you can provide us (via both your Network console, the Log Viewer utility, and `docker-compose logs -f web` if need be), the better.

Bear in mind we are also volunteers who create this software and then give it away for free, so we do not offer ETAs on issue resolution or any sort of service level agreement.

**Comment by DJMike67** [comment]
@SlvrEagle23 Many thanks for your reply. I totally appreciate that you are doing this on a voluntary basis for no monetary gain. I appreciate what you guys do for us end-users. Many thanks! 

I have done some testing, the steps and results of which are described below. This might help you to resolve my problem...

TEST1
Using FTP I uploaded an MP3 file
I waited several minutes but the "File Not Processing" message did not disappear
Using the web interface I selected the file and clicked the MOVE button
I selected the same folder that the file was already in - because I did not want to actually move the file but I read on another issue that clicking MOVE might kickstart the processing
I got a confirmation message that the file had been moved (even though it physically hadn't - it remained in the same folder)
RESULT: The File Not Processed message disappeared

TEST2
Using FTP I uploaded several more MP3's
I waited several minutes but the File Not Processing message did not disappear from any of them
Using the web interface I selected ALL the newly uploaded files and clicked the MOVE button
Again, I selected the same folder that the files were already in
I got a confirmation message that the files had been "moved"
RESULT: Only the first selected file was "moved". The File Not Processed message disappeared from that file but remained on the other files


TEST3
Using FTP I uploaded several more MP3's
I waited several minutes but the File Not Processing message did not disappear
Using the web interface I selected ALL files and clicked the MOVE button
This time I selected a different folder
I got a confirmation message that the files had been moved
I then navigated to the folder that the files had been moved to
RESULT: ALL selected files were moved and the File Not Processed message disappeared from them all however I then had to move them back to the folder that they were in originally

So yes there is a workaround to get the files processed (so that they can be added to playlists etc) however this is far from ideal, having to use the Move button and, if I want to do it in bulk, having to actually move them and move them back.

I have requested the log files from the company who provides my streaming server

**Comment by Vaalyn** [comment]
I've done some tests in my dev installation and the only way I can kinda reproduce this behaviour is by completely killing the "Media Sync Queue Worker". When I upload files when that is not running they show up as not processed and when I "move" the files to the same directory the first one selected is processed.

My suspicion is that maybe your hoster is running 1 AzuraCast installation for multiple users and overloading the available resources of the server or there are too many files on all those users for 1 "Queue Worker".

Could you ask your hoster if this is a shared AzuraCast installation, how the resource utilization of the server looks like and how many queue workers there are running?

----

A general question from my side: have you talked with your hoster about this issue (apart from getting those logs)?
Since they get paid by you they should be responsible for a) fixing problems with their infrastructure and b) when they can identify an issue that is due to a bug in the software itself, they should be co-ordinating with us instead of their paying users.

I'm asking this because many "cheap" hosters are trying to offload their support to our GitHub Repository so that they don't have to do that.

**Comment by DJMike67** [comment]
@Vaalyn Thanks for coming back to me. I have done some more testing. The result of which I will cover below. Regarding your other general question...You are 100% right. It is up to my hoster to provide support especially as I am paying them. I reported the issue to them but after extensive research into potential causes they were unable to provide a resolution and said that they would have to contact Azuracast developers. I did not see any post from them on Github (they have posted before) so in desparation I took matters into my own hands because it was having an impact on my ability to use AutoDJ and I opened a Github issue, So if that was the wrong thing to do I apologise.

@SlvrEagle23 - I used the F12 key in Chrome to capture the network log. I tested with one of the files that I knew was causing a problem. This file, although it appears to upload via web interface (the progress bar appears to complete), the file does not appear in the files list and if I upload via FTP, it appears in the files list but does not process. This is the result from the log (I have replaced the filename with XXXXX

 {"code":0,"message":"Warning for uploaded media file \"XXXXX\": [\"Invalid MPEG audio header (ff f8 67 ff ) at offset 262426\",\"Invalid MPEG audio header (00 00 00 08 ) at offset 262573\",\"Invalid MPEG audio header (ff ff ff ef ) at offset 262448\",\"Invalid MPEG audio header (ff ff ef 29 ) at offset 262449\",\"Invalid MPEG audio header (ff ef 29 0f ) at offset 262450\"]","formatted_message":"Warning for uploaded media file \"XXXXX\": [\"Invalid MPEG audio header (ff f8 67 ff ) at offset 262426\",\"Invalid MPEG audio header (00 00 00 08 ) at offset 262573\",\"Invalid MPEG audio header (ff ff ff ef ) at offset 262448\",\"Invalid MPEG audio header (ff ff ef 29 ) at offset 262449\",\"Invalid MPEG audio header (ff ef 29 0f ) at offset 262450\"]","extra_data":[],"success":false}

Even with my limited knowledge this indicates a problem with the actual MP3 file? There were 2 other files with similar problems. But..other MP3 files that were processed/encoded with my audio editor on the same day have uploaded successfully and the same file plays with no problems locally on my computer and the same file, uploaded to Centovacast plays via it's AutoDJ  (I have 2 radio stations, both with the same hoster, one uses Azuracast, the other uses Centovcast)

Once again, if you could provide any help it would be much appreciated. Is this a problem with the MP3? Is it a problem with Azuracast? Is it a problem with my hosting company's installation of Azuracast?

They are running Rolling Release #b31f892 (2020-10-13 4:50) • Docker • PHP 7.4

**Comment by Vaalyn** [comment]
> ... I did not see any post from them on Github (they have posted before) so in desparation I took matters into my own hands because it was having an impact on my ability to use AutoDJ and I opened a Github issue, So if that was the wrong thing to do I apologise.

Thank you for answering that question. It's definitely not wrong to seek for help from us, especially when there we get the debugging information that we need to be able to investigate. I'm seeking to understand how the "ecosystem" of AzuraCast hosters is currently operating and for me it looks like most do not have any sufficient technical knowledge to debug/diagnose issues so whenever they can't easily help their customers they stick their head in the sand and let the customer deal with it alone. There are some that come here to seek help with issues that their customers have but that's rather the exception than the norm.

---

Regarding the MP3 files that produce the error that you posted, would you mind sending me one of those files? I'd like to take a look at one of those directly. That error looks like the getID3 library couldn't correctly parse the file.

**Comment by DJMike67** [comment]
@Vaalyn I have uploaded 2 MP3's to a folder as they are approx 60MB each. You can download them from here: https://app.box.com/s/7t3u49zjrvgjqkq8nzmcg6cj4pxtmdcj

0001 is one of the files that cannot be uploaded via Azuracast front end and if I upload via FTP, says file not processed
0005 has no problem so I uploaded it for you for comparison

FYI 0002, 0003 and 0004 also have the same problem as 0001 but there's no point uploading them. If we can resolve the problem by looking at 0001 we should be able to resolve for all.

Thanks once again for your help

**Comment by DJMike67** [comment]
@Vaalyn I should add that they are podcast episodes and although I didn't do the editing of them, I've been told that the settings used for all the MP3's were the same

**Comment by Vaalyn** [comment]
@DJMike67 I'm not a professional when it comes to analyzing raw audio frames so I've just done some checks with tools I've downloaded. This is what I found:

I've run both files through [`Checkmate MP3 Checker`](http://checkmate.gissen.nl/) on the CLI which reported the following:

**.\mpck-0.21.exe -v \0001.mp3**
```
mpck:.\0001.mp3:262432: unknown layer (1-3 allowed)
mpck:.\0001.mp3:262452: unknown bitrate
    version                       MPEG v1.0
    layer                         3
    bitrate                       128000 bps
    samplerate                    44100 Hz
    frames                        162720
    time                          70:50.644
    unidentified                  393 b (0%)
        before 1st frame          393 b
        between frames            0 b
        after last frame          0 b
    stereo                        yes
    size                          66672 KiB
    ID3V1                         yes
    ID3V2                         yes
        size                      262144 B
    APEV1                         no
    APEV2                         no
    last frame
        offset                    68272462 b (0x411c14e)
        length                    418
    errors                        invalid header values
    result                        Bad
```

**.\mpck-0.21.exe -v \0005.mp3**
```
SUMMARY: .\0005.mp3
    version                       MPEG v1.0
    layer                         3
    bitrate                       128000 bps
    samplerate                    44100 Hz
    frames                        157012
    time                          68:21.537
    unidentified                  407 b (0%)
        before 1st frame          407 b
        between frames            0 b
        after last frame          0 b
    stereo                        yes
    size                          64279 KiB
    ID3V1                         yes
    ID3V2                         yes
        size                      196608 B
    APEV1                         no
    APEV2                         no
    last frame
        offset                    65821215 b (0x3ec5a1f)
        length                    418
    errors                        none
    result                        Ok
```

So it's not just the [`getID3`](https://github.com/JamesHeinrich/getID3) library that has issues with that file.

I then ran a simple ffmpeg convert to a new MP3 file to see if that fixes the issue.
This is the command that I ran (copied from [this StackOverflow post](https://stackoverflow.com/a/12952172/3292313)):
```
ffmpeg -i 0001.mp3 -vn -ar 44100 -ac 2 -b:a 128k 0001-fixed.mp3
```

When ffmpeg started converting the file I got the following output:
```
[mp3 @ 0x7fe59c812400] Skipping 407 bytes of junk at 262154.
[mp3 @ 0x7fe59c812400] Estimating duration from bitrate, this may be inaccurate
```

When I remove the `-vn` option so that the album cover is not stripped from the output I additionally get those lines:
```
[swscaler @ 0x7fa5ce500000] deprecated pixel format used, make sure you did set range correctly
[mp3 @ 0x7fa5cf008200] Frame rate very high for a muxer not efficiently supporting it.
Please consider specifying a lower framerate, a different muxer or -vsync 2
```

I'm able to upload those 2 resulting MP3 files to AzuraCast without any errors and they show up immediately with all data:

![image](https://user-images.githubusercontent.com/13745863/96480883-1a630700-123b-11eb-9cf4-33693135b421.png)


Without spending some time on learning how to analyze the bytes of audio files directly I can't say what kind of "junk" data there is on that file but it seems like the encoder that produced that file is sometimes producing "junk" data that shouldn't be how it currently is.

**Comment by DJMike67** [comment]
@VaalynMany thanks for going the extra mile on this one! The comment/question that I know will be asked when I report this back is... Is the problem with the file or Azuracast? These files play OK directly on my computer (via QuickTime player or VLC for example) and they upload and play from Centovacast AutoDJ. So although your file checker reports an invalid header value, is it a problem that is only being experienced with Azuracast?

Thanks again

**Comment by Vaalyn** [comment]
@DJMike67 The thing is, **players** have no issue with skipping junk in the headers. In theory Liquidsoap and IceCast should be able to play those files too. Where the issues begin is that in order to correctly queue the files, send metadata and all the other stuff that AzuraCast provides you with we need to analyze the media files. If the software can't do that because there is junk in there that let's the analyzer fail, then it can't be correctly used.

Since Centova is closed source software (encrypted with IonCube) I can't exactly say how they do their metadata analysis but as far as I remember they where using some really old PHP source code written back in PHP 4 times that they've just updated to work on newer versions (without actually modernizing it). Most likely they are using a software that is not written in PHP to analyze media files with some custom code they've written to interface with it which probably ignores those issues.

We are using the [getID3](https://www.getid3.org/) PHP library which has been around since even before Centova was released so it "might" also be possible that they are using an old version of that which "might" not have validation for junk data. There is really now way for me to know.

If the one who produced those files is arguing about "technically the files are fine since they can be played" this completely misses the mark. Multiple tools say that those files are not correctly encoded / contain stuff that they shouldn't so there "technically" is no arguing about it. Those files are violating the mpeg format spezification if multiple validators say they contain stuff that they shouldn't. It's no surprise that audio players still happily play files that violate the spec since the decoders that most players use have been around for a long time and have been build to accept most stuff without making a fuss.

In theory we could also use another software / library to do the analyzing (and other things that we use the getID3 lib for) that can ignore those issues. For example since FFmpeg is able to deal with those files the analyzer [FFprobe](https://ffmpeg.org/ffprobe.html) can also deal with those and there is also a [PHP library](https://github.com/PHP-FFMpeg/PHP-FFMpeg) to interface with FFmpeg. But this still doesn't mean that the files are correctly encoded (remember that even FFmpeg complains about the junk data in there). This would be a workaround, if done at all,  which I'm neither hinting at nor proposing with this comment.

To summarize my ramblings, yes the files are playable but still violate the mpeg specification. The library that we currently use to analyze media files can not ignore those issues and as far as I'm aware there are no plans to switch the library to some other library.

**Comment by DJMike67** [comment]
@Vaalyn - thank you once again for the detailed explanation. I'm learning a lot about under the hood of MP3 processing!

**Comment by SlvrEagle23** [comment]
@DJMike67 It's quite possible that the fixes applied recently to SFTP upload notification handling are also applicable to this problem. If your host updates to the latest version, please retry the uploads and verify that the processing issue persists; it should be completely resolved at this point.

**Comment by DJMike67** [comment]
@SlvrEagle23 Many thanks. They are showing as being on: Rolling Release #bc307d2 (2020-10-21 12:52) 

Is there a later version?

**Comment by SlvrEagle23** [comment]
@DJMike67 Yep, the latest commit is `02b92bf` which is from 10/25. The fix in particular is from `4c7fe5c` on the same day (10/25).

**Comment by DJMike67** [comment]
I'd like to close this issue now. It appears to be an issue with 4 specific files of mine. I'd like to thank you all for your help

Mike
